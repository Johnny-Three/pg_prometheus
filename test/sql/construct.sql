\set ECHO ALL
\set ON_ERROR_STOP 1
SET TIME ZONE 'UTC';

SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test'::text, 12.7::double precision, '{}'::jsonb);
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test'::text, 12.7::double precision, '{"key": "value"}'::jsonb);
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test2'::text, 12.7::double precision, '{"key1": "value1", "key2": "value2"}'::jsonb);

\set ON_ERROR_STOP 0
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test2'::text, 12.7::double precision, '{"key1": "value1", "key2": 4}'::jsonb);
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test2'::text, 12.7::double precision, '{"key1": "value1", "key2": {"key3":"value2"}}'::jsonb);
