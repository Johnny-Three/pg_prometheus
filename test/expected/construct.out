\set ECHO ALL
\set ON_ERROR_STOP 1
SET TIME ZONE 'UTC';
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test'::text, 12.7::double precision, '{}'::jsonb);
        prom_construct        
------------------------------
 test 12.700000 1494595898000
(1 row)

SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test'::text, 12.7::double precision, '{"key": "value"}'::jsonb);
              prom_construct               
-------------------------------------------
 test{key="value"} 12.700000 1494595898000
(1 row)

SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test2'::text, 12.7::double precision, '{"key1": "value1", "key2": "value2"}'::jsonb);
                       prom_construct                       
------------------------------------------------------------
 test2{key1="value1",key2="value2"} 12.700000 1494595898000
(1 row)

\set ON_ERROR_STOP 0
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test2'::text, 12.7::double precision, '{"key1": "value1", "key2": 4}'::jsonb);
ERROR:  Jsonb labels must be a set of string keys mapped to string values.
SELECT prom_construct(Timestamp with time zone 'Fri May 12 13:31:38 2017', 'test2'::text, 12.7::double precision, '{"key1": "value1", "key2": {"key3":"value2"}}'::jsonb);
ERROR:  Jsonb labels must be a set of string keys mapped to string values: cannot have nested labels.
